sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -37,6 +37,16 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSparkSession {
       ).toDF("name", "age", "height")
   }
 
+  def createNaNDF(): DataFrame = {
+    Seq[(java.lang.Integer, java.lang.Long, java.lang.Short,
+      java.lang.Byte, java.lang.Float, java.lang.Double)](
```

Comment:
Do we have to use boxed type? how about
```
Seq((1, 1L, 1.toShort, 1.toByte, 1.0f, 1.0), (0, 0L, ...)).toDF...
```

Reply:
I was following the rest of the file's conventions. Happy to update

Reply:
Just tried it this way and it fails with `sbt.ForkMain$ForkError: java.lang.NoClassDefFoundError: no Java class corresponding to Product with Serializable found`
So we do need to use boxed type

Reply:
OK we can keep the type declaration, but can we use `1L`, `1.toShort`, etc. when writing the values?

Reply:
Definitely can do so for the values, updated.